src/sinks/tcp.rs

```diff
@@ -50,7 +76,48 @@ impl SinkConfig for TcpSinkConfig {
             .next()
             .ok_or_else(|| "Unable to resolve DNS for provided address".to_string())?;
 
-        let sink = raw_tcp(addr, acker, self.encoding.clone());
+        let tls = match self.tls {
+            Some(ref tls) => {
+                if tls.enabled.unwrap_or(false) {
+                    if tls.key_file.is_some() != tls.crt_file.is_some() {
+                        return Err("Must specify both tls key_file and crt_file".into());
```

Comment:
I was thinking it might make sense to make these non optional items and treat tls as enabled iff TcpConfig.tls is set to Some. Then we can also provide an enabled option to disable tls with both key and crt set. What do you think?

Reply:
So there are two things I think you're saying here:

1. Drop the `tls.enabled` setting, and treat TLS as enabled if the `tls` config exists.
2. Make `tls.crt_file` and `tls.key_file` as required items.

Both of these are different than the spec, which is why I coded it the way it is. However I would tend to agree with the first point. The difficulty comes that it then requires some field be present to cause it to be `Some`, which is what the `tls.enabled` item provides with all the others being optional.

On the second, note that this is a client, and a certificate is not usually required on the client side unless the server is specifically authenticating the client.